### PR TITLE
CORE-1002: Support chain-specific BRCryptoHash encoding

### PR DESCRIPTION
**Pull request by ewerx**
Tezos uses Base58check encoding for transaction hash strings, other chains use hex encoding.

### CODE REVIEW

File: WalletKitCore/src/crypto/BRCryptoHash.c [hunk 1]
```diff
@@ -46,6 +49,11 @@ cryptoHashEqual (BRCryptoHash h1, BRCryptoHash h2) {
                                0 == memcmp (h1->bytes, h2->bytes, h1->bytesCount)));
 }
 
+extern OwnershipGiven char *
+cryptoHashEncodeString (BRCryptoHash hash) {
+    return cryptoNetworkEncodeHash (hash);
+}
+
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoHash.c:56]: This is needed?  It is a bit iffy for lower-level BRCryptoHash.c to include the higher-level BRCryptoNetworkP.h; okay, by iffy.

Comment (ewerx) [reply WalletKitCore/src/crypto/BRCryptoHash.c:56]: From the client (Swift) side this is called on a transaction hash with no visibility to the BRCryptoNetwork, so it was needed to maintain the interface. A BRCryptoNetwork is not given by the caller but given the chain type stored in the BRCryptoHashRecord the `cryptoNetworkEncodeHash` can look up the handler. I agree this workaround is a bit iffy, open to suggestions for a better solution that doesn't require call site to have network visibility.

Comment (EBGToo) [reply WalletKitCore/src/crypto/BRCryptoHash.c:56]: We can move forward with the circularity.  We should look at the Swift/Java use to see if the hash interface is used.  (Surely it is in the 'TransferView', even in Core Demo.)

File: WalletKitCore/src/crypto/handlers/btc/BRCryptoNetworkBTC.c [hunk 2]
```diff
@@ -192,8 +192,14 @@ cryptoNetworkInitializeAccountBTC (BRCryptoNetwork network,
...
-    BRCryptoHash hash; //TODO:XTZ
-    return hash;
+    assert(64 == strlen (string));
+    UInt256 hash = uint256(string);
+    return cryptoHashCreateAsBTC (hash);
+}
+
+static char *
+cryptoNetworkEncodeHashBTC (BRCryptoHash hash) {
+    return cryptoHashStringAsHex (hash);
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/handlers/btc/BRCryptoNetworkBTC.c:202]: This keeps the `0x` prefix?

**Review by EBGToo — COMMENTED** [review]
Shouldn't most of this be merged to refactor2020?

File: WalletKitCore/src/crypto/handlers/btc/BRCryptoNetworkBTC.c (hunk 2 above)
Comment (ewerx) [reply WalletKitCore/src/crypto/handlers/btc/BRCryptoNetworkBTC.c:202]: Yes

**Comment by ewerx** [comment]
replaced by !117

**Pull request closed without merging**